foundationdb/src/directory/node.rs

```diff
@@ -0,0 +1,65 @@
+use crate::tuple::{Subspace};
+use crate::future::FdbSlice;
+use crate::{Transaction, Directory, DirectoryResult};
+use crate::directory::directory_subspace::DirectorySubspaceResult;
+
+
+pub struct Node {
+    subspace: Subspace,
+    path: Vec<String>,
+    target_path: Vec<String>,
+    _layer: Option<Vec<u8>>
+}
+
+impl Node {
+
+    pub fn exists(&self) -> bool {
+        // if self.subspace == None {
+        //     return false;
+        // }
+
+        true
+    }
+
+    pub async fn prefetchMetadata(&self, trx: &Transaction) -> &Node {
+        if self.exists() {
+           self.layer(trx).await;
+        }
+
+        return self;
+    }
+
+    pub async fn layer(&mut self, trx: &Transaction) -> &[u8] {
+        if self._layer == None {
+            let key = self.subspace.subspace(&b"layer".to_vec());
+            self._layer = match trx.get(key.bytes(), false).await {
+                Ok(None) => Some(Vec::new()),
+                Err(_) => Some(Vec::new()),
```

Comment:
I don't think silencing an FdbError is ok.
layer should probably return a Result